internal/cmd/providers.go

```diff
@@ -36,100 +37,166 @@ func providersCmd() *cobra.Command {
+		},
+	}
 
-			if cemanagerURL.Scheme != ingressHost.Scheme {
-				return xerrors.Errorf("Coder access url and hostname must have matching protocols: coder access url: %s, workspace provider hostname: %s", cemanagerURL.String(), ingressHost.String())
-			}
+	cmd.Flags().BoolVar(&fromContext, "from-context", false, "use current kube context to retrieve cluster-cert, cluster-address, sa-token, and namespace if not set explicitly")
+	cmd.Flags().StringVar(&clusterCA, "cluster-cert", "", "kubernetes cluster certificate")
+	cmd.Flags().StringVar(&clusterAddr, "cluster-address", "", "kubernetes cluster address")
+	cmd.Flags().StringVar(&SAToken, "sa-token", "", "kubernetes service account token")
+	cmd.Flags().StringVar(&namespace, "namespace", "", "kubernetes namespace")
+	cmd.Flags().StringVar(&storageClass, "storage-class", "", "storage class name to use for workspaces")
+	cmd.Flags().BoolVar(&sshEnabled, "ssh-enabled", false, "enable ssh")
+	cmd.Flags().StringVar(&environmentSA, "workspace-sa", "", "kubernetes service account name to use for workspaces")
+	cmd.Flags().StringArrayVar(&orgAllowlist, "org-allowlist", nil, "list organization IDs to allowlist")
+	return cmd
+}
 
-			// ExactArgs(1) ensures our name value can't panic on an out of bounds.
-			createReq := &coder.CreateWorkspaceProviderReq{
-				Name:           args[0],
-				Type:           coder.WorkspaceProviderKubernetes,
-				Hostname:       hostname,
-				ClusterAddress: clusterAddress,
-			}
+func installProviderCmd() *cobra.Command {
+	var (
+		skipPrompt bool
+	)
+	cmd := &cobra.Command{
+		Use:   "install",
+		Short: "Install kubernetes resources for a new workspace provider.",
+		Long:  "Install kubernetes resources for a new workspace provider.",
+		Example: `# create a new workspace provider
 
-			wp, err := client.CreateWorkspaceProvider(ctx, *createReq)
+coder providers install`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				ctx = cmd.Context()
+				err error
+			)
+
+			kctx, err := xkube.CurrentKubeContext()
 			if err != nil {
-				return xerrors.Errorf("create workspace provider: %w", err)
+				return xerrors.Errorf("reading kubernetes context: %w", err)
+			}
+			fmt.Printf(`
+Current Kubernetes context details:
+
+Context Name: %s
+Namespace:    %s
+
+The following kubernetes resources will be created or updated:
+`, kctx.ContextName, kctx.Namespace)
+			fmt.Printf(`
+-- ServiceAccount
+Name: %s
+Namespace: %s
+`, xkube.ResourceName, kctx.Namespace)
+			fmt.Printf(`
+-- ServiceAccount
+Name: %s
+Namespace: %s
+`, xkube.WorkspaceSA, kctx.Namespace)
+			fmt.Printf(`
+-- Role
+Name: %s
+Namespace: %s
+Rules: 
+%s`, xkube.ResourceName, kctx.Namespace, xkube.PrettyRules())
+			fmt.Printf(`
+-- RoleBinding
+Role:
+  Kind:  Role
+  Name:  %s
+Subjects:
+  - Kind: ServiceAccount
+    Name: %s
+    Namespace: %s
+
+`, xkube.ResourceName, xkube.ResourceName, kctx.Namespace)
+
+			if !skipPrompt {
+				fmt.Println("Press the Enter Key to continue")
```

Comment:
a y/n confirmation might be better, because it makes it easy to exit (though people can also do ^C)